src/support/slack/commands/run-scrape.js

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import { hasText, isObject } from '@adobe/spacecat-shared-utils';
+
+import BaseCommand from './base.js';
+import { triggerScraperRun } from '../../utils.js';
+import {
+  extractURLFromSlackInput,
+  postErrorMessage,
+  postSiteNotFoundMessage,
+} from '../../../utils/slack/base.js';
+import { isValidDateInterval } from '../../../utils/date-utils.js';
+
+const PHRASES = ['run scrape'];
+
+/**
+ * Factory function to create the RunScrapeCommand object.
+ *
+ * @param {Object} context - The context object.
+ * @returns {RunScrapeCommand} The RunScrapeCommand object.
+ * @constructor
+ */
+function RunScrapeCommand(context) {
+  const baseCommand = BaseCommand({
+    id: 'run-scrape',
+    name: 'Run Scrape',
+    description: 'Runs the specified scrape type for the site identified with its id, and optionally for a date range.'
+            + '\nOnly selected SpaceCat fluid team members can run scraper.'
+            + '\nCurrently this will run the scraper for all sources and all destinations configured for the site, hence be aware of costs'
+            + ' (source: ahrefs) when choosing the date range.',
+    phrases: PHRASES,
+    usageText: `${PHRASES[0]} {baseURL} {startDate} {endDate}`,
+  });
+
+  const { dataAccess, log } = context;
+
+  /**
+     * Validates input and triggers a new scrape run for the given site.
+     *
+     * @param {string[]} args - The arguments provided to the command ([site]).
+     * @param {Object} slackContext - The Slack context object.
+     * @param {Function} slackContext.say - The Slack say function.
+     * @returns {Promise} A promise that resolves when the operation is complete.
+     */
+  const handleExecution = async (args, slackContext) => {
+    const { say, user } = slackContext;
+    const config = await dataAccess.getConfiguration();
+    const slackRoles = config.getSlackRoles() || {};
+    const admins = slackRoles?.scrape || [];
+
+    if (!admins.includes(user)) {
+      await say(':error: Only selected SpaceCat fluid team members can run scraper.');
```

Comment:
not only fluid team members... i'd adjust to something like "Only members of role "scrape" can run this command"